### PR TITLE
fix smart pipe mapping helper runtiming

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
fixes it runtiming when you use it for mapping

fixes ported from this pr https://github.com/BeeStation/BeeStation-Hornet/pull/9220
## Why It's Good For The Game
you can now use it for mapping
## Changelog
:cl:
fix: smart pipe mapping helper runtiming
/:cl:

### CODE REVIEW

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -185,28 +185,44 @@
...
+	for(var/obj/effect/mapping_helpers/simple_pipes/helper in self_turf.contents)
+		if(helper == src)
+			continue
+		if(helper.piping_layer != src.piping_layer)
+			continue
+		if(helper.crashed)
+			return
+		CRASH("Duplicate simple_pipes mapping helper at [AREACOORD(src)]")
+		helper.crashed = TRUE
+		src.crashed = TRUE
```
Comment (Kapu1178) [inline code/modules/mapping/mapping_helpers.dm:206]: lol....

CRASH() ends the proc. This code is unreachable.

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -185,28 +185,44 @@
...
+	var/turf/self_turf = get_turf(src)
+	for(var/obj/effect/mapping_helpers/simple_pipes/helper in self_turf.contents)
+		if(helper == src)
+			continue
+		if(helper.piping_layer != src.piping_layer)
+			continue
+		if(helper.crashed)
+			return
+		helper.crashed = TRUE
+		src.crashed = TRUE
```
Comment (TiviPlus) [inline code/modules/mapping/mapping_helpers.dm:205]:
```suggestion
		crashed = TRUE
```

File: code/modules/mapping/mapping_helpers.dm [hunk 3]
```diff
@@ -185,28 +185,44 @@
 	var/pipe_color = ""
 	var/connection_num = 0
 	var/hide = FALSE
+	/// Tracking variable to prevent duplicate runtime messages
+	var/crashed = FALSE
 
 /obj/effect/mapping_helpers/simple_pipes/LateInitialize()
-	var/list/connections = list( dir2text(NORTH) = FALSE, dir2text(SOUTH) = FALSE , dir2text(EAST) = FALSE , dir2text(WEST) = FALSE)
+	var/list/connections = list( dir2text(NORTH)  = FALSE, dir2text(SOUTH) = FALSE , dir2text(EAST) = FALSE , dir2text(WEST) = FALSE)
```
Comment (TiviPlus) [inline code/modules/mapping/mapping_helpers.dm:192]: random space adding

**Review by TiviPlus — COMMENTED** [review]
Pending other pr

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**